util/internal/integrationtest/backend/elasticsearch/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/elasticsearch/dialect/ElasticsearchVersionUtils.java

```diff
@@ -31,57 +31,69 @@ private static int compare(ElasticsearchVersion a, ElasticsearchVersion b, int d
 				.compare( a, b );
 	}
 
-	public static boolean isOpenSearch(ElasticsearchVersion actual) {
-
-		return isDistribution( actual, ElasticsearchDistributionName.OPENSEARCH );
-	}
-
-	private static boolean isDistribution(
+	public static boolean condition(
 			ElasticsearchVersion actual,
-			ElasticsearchDistributionName distribution
+			Predicate<ElasticsearchVersionCondition> elasticsearchPredicate,
+			Predicate<ElasticsearchVersionCondition> opensearchPredicate
 	) {
-
-		return actual.distribution().equals( distribution );
+		ElasticsearchVersionCondition condition = new ElasticsearchVersionCondition( actual );
+
+		switch ( actual.distribution() ) {
+			case ELASTIC:
+				return elasticsearchPredicate.test( condition );
+			case OPENSEARCH:
+				return opensearchPredicate.test( condition );
+			default:
+				throw new IllegalStateException( "Unknown distribution" );
+		}
 	}
 
-	public static boolean isMatching(ElasticsearchVersion actual, String version) {
-		ElasticsearchVersion v = ElasticsearchVersion.of( version );
+	public static class ElasticsearchVersionCondition {
```

Comment:
Nice idea, that way we are forced to always go through `condition` to test a version.

I'd go a step further: you could easily get rid of `ElasticsearchVersionUtils` by moving all its methods into `ElasticsearchVersionCondition` and making that class top-level.